PR: https://github.com/odoo/odoo/pull/149164

From: 643594abd17ddd201bab838a68b4f5272201f018
From: Yash Vaishnav
Date: 2024-05-14 08:41:47

Structural Changes: 4
Total Changes: 7

[IMP] hr,website_slides: send notification to the user of data change

In this commit,
- The many2many field is created in the settings of the employee
- Those users will receive a notification of the data change of the employee
- It'll send the notification only when employee editing is enabled

task-3642351

closes odoo/odoo#149164

Related: odoo/enterprise#54184
Signed-off-by: Sofie Gvaladze (sgv) <sgv@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr/models/res_company.py
+++ b/addons/hr/models/res_company.py
@@ -10,3 +10,4 @@ class Company(models.Model):
     hr_presence_control_email_amount = fields.Integer(string="# emails to send")
     hr_presence_control_ip_list = fields.Char(string="Valid IP addresses")
     employee_properties_definition = fields.PropertiesDefinition('Employee Properties')
+    hr_notify_user_ids = fields.Many2many('res.users')

--- a/addons/hr/models/res_config_settings.py
+++ b/addons/hr/models/res_config_settings.py
@@ -21,6 +21,7 @@ class ResConfigSettings(models.TransientModel):
     hr_presence_control_email_amount = fields.Integer(related="company_id.hr_presence_control_email_amount", readonly=False)
     hr_presence_control_ip_list = fields.Char(related="company_id.hr_presence_control_ip_list", readonly=False)
     hr_employee_self_edit = fields.Boolean(string="Employee Editing", config_parameter='hr.hr_employee_self_edit')
+    hr_notify_user_ids = fields.Many2many(related='company_id.hr_notify_user_ids', readonly=False)
 
     @api.constrains('module_hr_presence', 'hr_presence_control_email', 'hr_presence_control_ip')
     def _check_advanced_presence(self):

--- a/addons/hr/models/res_users.py
+++ b/addons/hr/models/res_users.py
@@ -225,6 +225,11 @@ class User(models.Model):
 
     def _get_personal_info_partner_ids_to_notify(self, employee):
         # To override in appropriate module
+        if self.env['ir.config_parameter'].sudo().get_param('hr.hr_employee_self_edit'):
+            return (
+                'You were receving this message because you are set as notifier',
+                self.env.companies.hr_notify_user_ids.partner_id.ids
+            )
         return ('', [])
 
     def write(self, vals):
